run_project_tests.py

```diff
@@ -105,6 +105,11 @@ def stop_handler(signal, frame):
 signal.signal(signal.SIGINT, stop_handler)
 signal.signal(signal.SIGTERM, stop_handler)
 
+def setup_pythonpath():
+    # Make sure python can import mesonbuild.
+    test_dir = os.path.dirname(os.path.abspath(__file__))
+    os.environ['PYTHONPATH'] = test_dir
```

Comment:
Yes, I agree it would regress #3015. I fixed it up and tested both in-tree and installed, out-of-tree tests, and they are both working now.

What I did is to `import mesonbuild` and add the containing directory into `PYTHONPATH` as an absolute path. In an in-tree build, this will be `.` in preference over an installed meson, so everything works. For an out-of-tree build, this will be the installed meson, so everything works.

Using this approach, we don't have to guess or make heuristics about how to locate meson; we can just import it and let python do the hard work. `mesonbuild` should always be importable by these tests anyway, as we don't a bunch of imports from it at the top of the script. So, if we can't even import `mesonbuild`, the tests will fail before they hit this code.